tuxemon/event/conditions/check_evolution.py

```diff
@@ -19,118 +23,74 @@ class CheckEvolutionCondition(EventCondition):
-                            elif (
-                                _method == EvolutionType.variable
-                                and evolution.variable
-                            ):
-                                parts = evolution.variable.split(":")
-                                key = parts[0]
-                                value = parts[1]
-                                if (
-                                    key in player.game_variables
-                                    and player.game_variables[key] == value
-                                ):
-                                    evolving.append((monster, evolved))
-                                    _evolution = True
+                    op = "greater_or_equal"
+                    if compare(op, monster.level, evolution.at_level):
+                        _evolution = True
+                    if evolution.gender == monster.gender:
+                        _evolution = True
+                    if character.has_type(evolution.element):
+                        _evolution = True
+                    if character.has_tech(evolution.tech):
+                        _evolution = True
+                    if evolution.inside == client.map_inside:
+                        _evolution = True
+                    if evolution.traded == monster.traded:
+                        _evolution = True
+                    if evolution.stats:
+                        params = evolution.stats.split(":")
+                        operator = params[1]
+                        stat1 = monster.return_stat(StatType(params[0]))
+                        stat2 = monster.return_stat(StatType(params[2]))
+                        _evolution = compare(operator, stat1, stat2)
+                    if evolution.variable:
+                        parts = evolution.variable.split(":")
+                        key = parts[0]
+                        value = parts[1]
+                        if (
+                            key in character.game_variables
+                            and character.game_variables[key] == value
+                        ):
+                            _evolution = True
+                    if evolution.steps:
+                        result = evolution.steps - int(monster.steps)
+                        if result == 0:
+                            _evolution = True
+                            monster.steps += 1
+                        monster.levelling_up = True
+                        monster.got_experience = True
+                    # use the item on the monster
+                    if evolution.item:
+                        _evolution = False
+                    if _evolution:
+                        evolving.append((monster, evolved))
+
         if evolving:
-            player.pending_evolutions = evolving
-        return _evolution
+            character.pending_evolutions = evolving
+        return bool(evolving)
```

Comment:
I would instead check whether the list (`evolving`) is empty or not. I feel like it would be more descriptive of what is verified here.